apps/dashboard/src/routes/courses/[id]/lessons/+page.svelte

```diff
@@ -81,6 +87,47 @@
       return index + 1;
     }
   }
+
+  function findFirstIncompleteLesson() {
+    return $lessons.find(
+      (lesson) => lesson.lesson_completion.length === 0 && lesson.is_unlocked === true
+    );
+  }
+
+  onMount(() => {
+    // Update loaded lessons when $lessons store changes
+    lessons.subscribe((value) => {
+      loadedLessons = value;
+      // Check if it's the initial load
+      if (isInitialLoad) {
+        isInitialLoad = false;
+      }
+    });
+  });
```

Comment:
@tunny17 I don't understand why you need this.

This code is more complex.

Reply:
I added this basically because this ```goto(`/courses/${data.courseId}/lessons/${incompleteLesson.id}`);``` function still doesn't have access to the updated $lessons store so i did that to update the isInitialLoad variable to catch the change then run the goto function